plugins/src/main/java/org/smartregister/fhir/gateway/plugins/LocationHierarchyEndpointHelper.java

```diff
@@ -443,23 +453,25 @@ public List<String> generateAdminLevels(
 
     public List<Location> filterLocationsByAdminLevels(
             List<Location> locations, List<String> postFetchAdminLevels) {
-        if (postFetchAdminLevels == null) {
+
+        if (postFetchAdminLevels == null || postFetchAdminLevels.isEmpty()) {
```

Comment:
Resolves bug of filtering out all locations when adminlevels is not provided/is empty